### PR TITLE
story/FCT2-21031-Maintenenace-Page

### PR DESCRIPTION
**Pull request by renjithabby**
# PR checklist

Tick what applies before you raise. Delete anything that doesn't.

## Correctness
- [x] Does what the ticket asks for
- [x] Handles the edge cases (empty, null, zero, large inputs), not just the happy path
- [x] Errors are handled, not swallowed
- [x] New logic has tests, including the failure cases
- [x] Tests assert something real, not just run the code

## Keeping it simple
- [x] Doesn't rebuild something that already exists in the codebase
- [x] No more complex or slower than it needs to be
- [x] No leftover debug logging or commented-out code

## Easy to miss (a green pipeline won't flag these)
- [x] One logical change, not a pile of unrelated stuff

### If you touched the UI
- [x] Ran the e2e/ suite locally (the PR build only runs the mocked tests; the full suite runs post-merge and gates the deploy)
- [x] API changes are reflected in the MSW handlers in src/mocks and the matching zod schema, so the mocked tests aren't passing against a contract that's gone
- [x] New config or feature flags are wired into the VITE build variables, not just added in code
- [x] Data fetches handle the loading, empty and error states, not just success, and match how the rest of the app already does it
- [x] Ran prettier (npm run prettier:format), since CI lints but doesn't check formatting
- [x] Checked accessibility (keyboard nav, screen reader, sensible labels), since the Lighthouse run is manual and won't flag it for you

### CODE REVIEW

File: ui-spa/src/components/AppRoutes.tsx [hunk 1]
```diff
@@ -36,7 +38,10 @@ const AppRoutes = () => {
         },
       });
     }
-  }, [featureFlagsData, featureFlags, dispatch]);
+    if (featureFlagsData?.maintenanceMode) {
```
Comment (HasanCPS) [inline ui-spa/src/components/AppRoutes.tsx]: ####  Redirect fires on every render and there is no way to exit maintenance mode

The `useEffect` at line 41 navigates to `/maintenance` every time `featureFlagsData` changes and `maintenanceMode` is truthy. If the user is already on `/maintenance` and anything causes a re-render it will navigate again. More importantly, when the flag flips back to `false` the user stays stuck on `/maintenance` because there is nothing to send them back out.

```tsx
// Lines 41-43 — currently:
if (featureFlagsData?.maintenanceMode) {
  navigate("/maintenance", { replace: true });
}

// Suggested fix:
if (featureFlagsData?.maintenanceMode && location.pathname !== "/maintenance") {
  navigate("/maintenance", { replace: true });
}
if (!featureFlagsData?.maintenanceMode && location.pathname === "/maintenance") {
  navigate("/", { replace: true });
}
```

File: ui-spa/src/components/AppRoutes.tsx [hunk 2]
```diff
@@ -46,6 +51,7 @@ const AppRoutes = () => {
         path="/case/:caseId/case-management"
         element={<CaseManagementPage />}
       />
+      <Route path="/maintenance" element={<MaintenancePage />} />
```
Comment (HasanCPS) [inline ui-spa/src/components/AppRoutes.tsx:61]: #### `/maintenance` route is accessible to anyone regardless of whether maintenance mode is on

```tsx
<Route path="/maintenance" element={<MaintenancePage />} />
```

This sits outside `<ProtectedRoutes />` (line 56). Any user can navigate to `/maintenance` directly and see "Sorry this service is unavailable" even when the service is running fine. If someone bookmarks it or shares the link, they get a false outage message.

Either move it inside `<ProtectedRoutes />` with a conditional render based on the flag, or add a redirect inside `MaintenancePage` that sends users to `/` when `FEATURE_FLAG_MAINTENANCE_MODE` is `false`.

Comment (renjithabby) [reply ui-spa/src/components/AppRoutes.tsx:61]: wouldnt  the fix above  (!featureFlagsData?.maintenanceMode && location.pathname === "/maintenance") {
  navigate("/", { replace: true });
}  fix this issue

Comment (renjithabby) [reply ui-spa/src/components/AppRoutes.tsx:61]: the fist comment fix , will fix the second issue , added integration test for this scenario

**Review by HasanCPS — APPROVED** [review]
LGTM!
